alerting/src/main/kotlin/org/opensearch/alerting/resthandler/RestExecuteWorkflowAction.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.resthandler
+
+import org.apache.logging.log4j.LogManager
+import org.opensearch.alerting.AlertingPlugin
+import org.opensearch.alerting.action.ExecuteWorkflowAction
+import org.opensearch.alerting.action.ExecuteWorkflowRequest
+import org.opensearch.client.node.NodeClient
+import org.opensearch.common.unit.TimeValue
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.alerting.model.Workflow
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.rest.BaseRestHandler
+import org.opensearch.rest.RestHandler
+import org.opensearch.rest.RestRequest
+import org.opensearch.rest.action.RestToXContentListener
+import java.time.Instant
+
+private val log = LogManager.getLogger(RestExecuteWorkflowAction::class.java)
+
+class RestExecuteWorkflowAction : BaseRestHandler() {
+
+    override fun getName(): String = "execute_workflow_action"
+
+    override fun routes(): List<RestHandler.Route> {
+        return listOf(
+            RestHandler.Route(RestRequest.Method.POST, "${AlertingPlugin.WORKFLOW_BASE_URI}/{workflowID}/_execute")
+        )
+    }
+
+    override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        log.debug("${request.method()} ${AlertingPlugin.WORKFLOW_BASE_URI}/_execute")
+
+        return RestChannelConsumer { channel ->
+            val dryrun = request.paramAsBoolean("dryrun", false)
+            val requestEnd = request.paramAsTime("period_end", TimeValue(Instant.now().toEpochMilli()))
```

Comment:
Is there another param for this called, `period_end`? If so, what is it exactly for?

Reply:
[it's only for execute request purpose to build schedule
](https://github.com/opensearch-project/alerting/blob/0d778b8e875975c67ffbf539d9cac87e6bcf54c9/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteWorkflowAction.kt#L62-L63)

Reply:
https://github.com/opensearch-project/alerting/blob/0d778b8e875975c67ffbf539d9cac87e6bcf54c9/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteWorkflowAction.kt#L62-L63

Reply:
sounds good